### ISSUE TITLE
T80 of MSX_MiST and the ESP8266 Wi-Fi BIOS are not compatible

### ISSUE THREAD

**Issue by mist-user** [mist-devel/MSX_MiST#7]
KdL gave me some feedback regarding the T80 cpu, thanks!

I am testing MSX_MiST's T80 cpu and ran into this problem: this T80 prevents using the ESP8266 Wi-Fi BIOS.
The ESP_RESET routine comes out with error displaying the string “ESP8266 Not Found”.
So, since there are many changes in T80 of MSX_MiST, I ask if you can inform the author, maybe he wants to take a look at it.
 
[esp8266x_20220813-2231_full_bios.zip](https://github.com/mist-devel/MSX_MiST/files/11353548/esp8266x_20220813-2231_full_bios.zip)

Can you help KdL please?

**Comment by mist-user** [comment]
You need this special OCM-BIOS.DAT version with Nextor support and with embedded driver for Wi-Fi.

**Comment by gnogni** [comment]
I tested some different variants of T80 cores that you modified here on GitHub. At the end of the test I could see that the ESP8266 BIOS problem was not present before the change you commit on March 31, 2021. So for now I think I will use T80-f7f776b-master with T80a-0917cdf restored.

[https://gnogni.altervista.org/ocm/misc/T80-f7f776b-master_with_T80a-0917cdf_restored.zip](url)

**Comment by gyurco** [comment]
Thanks for testing.
You mean this one causes the problem? Can you revert only this commit?
https://github.com/mist-devel/T80/commit/8cebad773e21001b131e6b4fc0f21c31edcca5b1

It's needed because of an issue found on ZX Next with interrupt vector latching at the wrong time. If this is the issue on MSX, then maybe the IM vector is not put on the bus at the right time, however it would be easy to check it if a real ESP is avaliable.

**Comment by gyurco** [comment]
...or you mean this one?
https://github.com/mist-devel/T80/commit/0917cdf114034702dab18db176e1adc4c761df0f
I don't see how the RESET_ESP routine is related to NMI.

**Comment by gnogni** [comment]
Yes, today's test confirmed that "T80a.vhd" is the cause of the ESP8266 BIOS issue, so I restored the entire file to before it was changed to ZX Spectrum Next. All subsequent variants do not allow the ESP8266 BIOS to be used, I tested them one by one
(ps. if you reread what I had written, you will see that it says: "T80-f7f776b-master" with "T80a-0917cdf" restored).
Regarding RESET_ESP I will send you privately the email of Ducasp who designed it.
